pkg/kv/kvserver/batcheval/cmd_gc.go

```diff
@@ -178,6 +182,18 @@ func GC(
 		return result.Result{}, err
 	}
 
+	// Fast path operation to try to remove all user key data from the range.
+	if rk := args.ClearRangeKey; rk != nil {
+		if !rk.StartKey.Equal(desc.StartKey.AsRawKey()) || !rk.EndKey.Equal(desc.EndKey.AsRawKey()) {
+			return result.Result{}, errors.Errorf("gc with clear range operation could only be used on the full range")
```

Comment:
This is going to happen with range merges. The GCer will fall back to point GC, right? We may want to return a structured error and retry it, although I suppose we'd have to bump it all the way back up to the GC queue and wait for it to run the merged range, in which case we could run the risk of never GCing the range if we, for some reason, ended up in a retry loop.

This doesn't seem very likely to happen, since most ranges will have cooled down from e.g. load-based splits by the time the GC TTL hits, so I don't think it's a huge deal.